File: booking-kit/src/main/java/com/skedgo/android/tripkit/booking/model/StringFormField.java
package com.skedgo.android.tripkit.booking.model;

import android.os.Parcel;
import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

public class StringFormField extends FormField {

  public static final Creator<StringFormField> CREATOR = new Creator<StringFormField>() {
    @Override
    public StringFormField createFromParcel(Parcel in) {
      in.readInt();
      return new StringFormField(in);
    }

    @Override
    public StringFormField[] newArray(int size) {
      return new StringFormField[size];
    }
  };
  @SerializedName("keyboardType")
  private String keyboardType;
  @SerializedName("value")
  private String value;

  public StringFormField(Parcel in) {
    super(in);
    this.keyboardType = in.readString();
    this.value = in.readString();
  }

  public StringFormField() {
    super();
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    dest.writeInt(STRING);
    super.writeToParcel(dest, flags);
    dest.writeString(keyboardType);
    dest.writeString(value);
  }

  public String getKeyboardType() {
    return keyboardType;
  }

  public void setKeyboardType(String keyboardType) {
    this.keyboardType = keyboardType;
  }

  @Nullable
  public String getValue() {
    return value;
  }

  public void setValue(String value) {
    this.value = value;
  }
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/StringFormField.kt
package com.skedgo.tripkit.booking

import android.os.Parcel
import android.os.Parcelable.Creator
import com.google.gson.annotations.SerializedName

class StringFormField : FormField {

    @SerializedName("keyboardType")
    var keyboardType: String? = null

    @SerializedName("value")
    var mValue: String? = null
    constructor(parcel: Parcel) : super( parcel) {  // Call the parent class constructor
        keyboardType = parcel.readString()  // Read String for keyboardType
        mValue = parcel.readString()  // Read String for value
    }

    // Secondary empty constructor (the default constructor)
    constructor() : super() {
        // Initialize any default values or leave it empty
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(STRING)
        super.writeToParcel(dest, flags)
        dest.writeString(keyboardType)
        dest.writeString(mValue)
    }

    override fun getValue(): String? {
        return mValue
    }

    fun setValue(value: String) {
        this.mValue = value
    }

    companion object CREATOR : Creator<StringFormField> {
        override fun createFromParcel(parcel: Parcel): StringFormField {
            parcel.readInt() // Read STRING type
            return StringFormField(parcel)
        }

        override fun newArray(size: Int): Array<StringFormField?> {
            return arrayOfNulls(size)
        }
    }
}
